### DISCUSSION TITLE
How to change Text for No records to display.

### DISCUSSION THREAD

**Discussion by sagrawal9689** [KevinVandy/material-react-table#425] [Q&A]
How do I change the text shown in body when there is no data in the table. Also Can I render my own jsx instead of the default text.

**Comment by KevinVandy** [comment]
localization prop

**Reply by sagrawal9689** [reply]
@KevinVandy But I can only change the text how can I render my own jsx instead.

**Reply by KevinVandy** [reply]
you can also render a custom table body with `muiTableBodyProps.children` for advanced render overrides like that. Only pass the prop when your `data.length === 0` though.

There was an example here: https://www.material-react-table.dev/?path=/story/styling-custom-table-body-examples--custom-table-body

**Comment by petrgazarov** [comment]
Should this be properly documented? Better yet, a new prop that does this more easily. Something like `emptyStateComponent`.
